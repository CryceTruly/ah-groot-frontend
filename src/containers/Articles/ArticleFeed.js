import React, { Component } from 'react';
import M from 'materialize-css/dist/js/materialize.js';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import AllArticleView from '../../components/articles/AllArticleView';
import { getAllArticles } from '../../actions/ArticleAction';
import Header from '../../components/landingPage/Header';

import '../../styles/viewarticle.scss';

export class ArticleFeed extends Component {
  constructor(props) {
    super(props);
    let { history } = this.props;
  }
  componentDidMount() {
    var elems = document.querySelectorAll('.fixed-action-btn');
    const options = {
      direction: 'right',
      hoverEnabled: false
    };
    M.FloatingActionButton.init(elems, options);
    this.props.getAllArticles();
  }


  render() {
    return (
      <div>
        <Header />
        <div className="container">
        <div className="fixed-action-btn">
          <a className="btn-floating btn-large">
            <i className="large material-icons">more_horiz</i>
          </a>
          <ul>
            <li onClick={()=>this.props.history.push('/me/articles')}>
              <a className="btn-floating green">
                <i className="material-icons">list</i>
              </a>
            </li>
            <li onClick={()=>this.props.history.push('/new-article')}>
              <a className="btn-floating blue">
                <i className="material-icons">mode_edit</i>
              </a>
            </li>
          </ul>
        </div>
          <div className="center">Your article feed</div>
          <div className="row">
            <div className="col s12">
              {this.props.allArticles.length > 0 ? (
                this.props.allArticles.map(element => {
                  return (
                    <AllArticleView
                      key={element.slug}
                      title={element.title}
                      description={element.description}
                      slug={element.slug}
                      average_rating={element.average_rating}
                    />
                  );
                })
              ) : (
                <span ></span>
              )}

            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return state;
};

const mapDispatchToProps = dispatch => {
  return {
    getAllArticles: function() {
      dispatch(getAllArticles());
    },
  };
};

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(ArticleFeed));


