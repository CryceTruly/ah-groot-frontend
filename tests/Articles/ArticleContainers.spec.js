import React from 'react';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import thunk from "redux-thunk";
import { Provider } from 'react-redux';
import moxios from 'moxios';

import ArticleFeed, { ArticleFeed as DumpArticleFeed} from '../../src/containers/Articles/ArticleFeed';
import data from '../landing_page/maxios_mock';
import SelectCategory, { SelectCategory as DumpCategory, mapDispatchToProps } from '../../src/containers/Articles/Category';


describe('All Article container:', () => {
    const initialState = {
        articles : data.article.articles.results
      };
      const mockStore = configureStore([thunk]);
      let store;

      beforeEach(() => {
        store = mockStore(initialState);
        moxios.install();
      });

      afterEach(function () {
        moxios.uninstall();
      });

    it('ArticleFeed should render without crashing', () => {
    const wrapper = shallow(<DumpArticleFeed
    allArticles ={data.article.articles.results}
    getAllArticles={()=>jest.fn()}
     />);
    expect(wrapper).toMatchSnapshot();
  });

  it('ArticleFeed should mount without crashing', () => {
    const wrapper = mount(<Router><DumpArticleFeed
    allArticles ={data.article.articles.results}
    getAllArticles={()=>jest.fn()}
     /></Router>);
     const push = jest.fn();
     wrapper.setProps({ history: { push } })
    expect(wrapper).toMatchSnapshot();
  });

  it('ArticleFeed should render with store without crashing', () => {
     const wrapper = mount(<Router><Provider store={store}><ArticleFeed
        allArticles ={data.article.articles.results}
        getAllArticles={()=>jest.fn()}
     /></Provider> </Router>);
     wrapper.update();
     expect(wrapper).toMatchSnapshot();
  });

  it('ArticleFeed should navigate to articles page', () => {
    const wrapper = mount(<Router><DumpArticleFeed
    allArticles ={data.article.articles.results}
    getAllArticles={()=>jest.fn()}
    history = {{push: jest.fn()}}
     /></Router>);
     wrapper.find('li').at(1).at(0).simulate('click');
  });

  it('ArticleFeed should navigate to edit articles page', () => {
    const wrapper = mount(<Router><DumpArticleFeed
    allArticles ={data.article.articles.results}
    getAllArticles={()=>jest.fn()}
    history = {{push: jest.fn()}}
     /></Router>);
     wrapper.find('li').last().simulate('click');
  });
});

describe('SelectCategory', () => {
    const initialState = {
        articles : data.article.articles.results
      };
      const mockStore = configureStore([thunk]);
      let store;

      beforeEach(() => {
        store = mockStore(initialState);
        moxios.install();
      });

      afterEach(function () {
        moxios.uninstall();
      });


      it('should render without crashing', () => {
        const wrapper = shallow(<DumpCategory
            allCategories ={data.category.categorys.results}
            getAllCategories={()=>jest.fn()}
            setCategory ={()=>jest.fn()}
         />);
        expect(wrapper).toMatchSnapshot();
      });

      it('should render with store without crashing', () => {
        const wrapper = mount(<Provider store={store}><SelectCategory
            allCategories ={data.category.categorys.results}
            getAllCategories={()=>jest.fn()}
        /></Provider>);
        wrapper.update();
        expect(wrapper).toMatchSnapshot();
     });

     it('should render with store without default category', () => {
      const wrapper = mount(<Provider store={store}><SelectCategory
          allCategories ={data.category.categorys.results}
          getAllCategories={()=>jest.fn()}
          defaultCategory={{'slug':'slug', "name":"name"}}
      /></Provider>);
      expect(wrapper).toMatchSnapshot();
   });

   it('should respond to on change events', () => {
    const wrapper = shallow(<DumpCategory
        allCategories ={data.category.categorys.results}
        getAllCategories={()=>jest.fn()}
        setCategory ={()=>jest.fn()}
        onChangeCategory={()=>jest.fn()}
     />);
   const event = {slug: "spam", name: 'spam'};

    wrapper.find('.basic-single').simulate('change', event);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match state to props', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setCategory();
 });
});

