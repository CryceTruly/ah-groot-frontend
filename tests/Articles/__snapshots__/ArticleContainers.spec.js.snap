// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Article container: ArticleFeed should mount without crashing 1`] = `ReactWrapper {}`;

exports[`All Article container: ArticleFeed should render with store without crashing 1`] = `ReactWrapper {}`;

exports[`All Article container: ArticleFeed should render without crashing 1`] = `ShallowWrapper {}`;

exports[`SelectCategory should render with store without crashing 1`] = `ReactWrapper {}`;

exports[`SelectCategory should render with store without default category 1`] = `ReactWrapper {}`;

exports[`SelectCategory should render without crashing 1`] = `ShallowWrapper {}`;

exports[`SelectCategory should respond to on change events 1`] = `ShallowWrapper {}`;
